jscomp/test/js_date_test.ml

```diff
@@ -207,224 +196,204 @@ let suites = Mt.[
 
     "setDate", (fun _ ->
       let d = date () in
-      let _ = N.setDate d 12. in
+      let _ = N.setDate 12. d in
 
       Eq(12., N.getDate d)
     );
     "setFullYear", (fun _ ->
       let d = date () in
-      let _ = N.setFullYear d 1986. in
+      let _ = N.setFullYear 1986. d in
 
       Eq(1986., N.getFullYear d)
     );
     "setFullYearM", (fun _ ->
       let d = date () in
-      let _ = N.setFullYearM d ~year:1986.
-                                     ~month:7. () in
+      let _ = N.setFullYearM d ~year:1986. ~month:7. in
       Eq( (1986., 7.),
           (N.getFullYear d,
            N.getMonth d))
     );
     "setFullYearMD", (fun _ ->
       let d = date () in
-      let _ = N.setFullYearMD d ~year:1986.
-                                      ~month:7.
-                                      ~date:23. () in
+      let _ = N.setFullYearMD d ~year:1986. ~month:7. ~date:23. in
       Eq( (1986., 7., 23.),
           (N.getFullYear d,
            N.getMonth d,
            N.getDate d))
     );
     "setHours", (fun _ ->
       let d = date () in
-      let _ = N.setHours d 22. in
+      let _ = N.setHours 22. d in
 
       Eq(22., N.getHours d)
     );
     "setHoursM", (fun _ ->
       let d = date () in
-      let _ = N.setHoursM d ~hours:22.
-                                  ~minutes:48. () in
+      let _ = N.setHoursM d ~hours:22. ~minutes:48. in
       Eq( (22., 48.),
           (N.getHours d,
            N.getMinutes d))
     );
     "setHoursMS", (fun _ ->
       let d = date () in
-      let _ = N.setHoursMS d ~hours:22.
-                                   ~minutes:48.
-                                   ~seconds:54. () in
+      let _ = N.setHoursMS d ~hours:22. ~minutes:48. ~seconds:54. in
```

Comment:
Cool that these keep working even if `d` is passed first :)

Reply:
absolutely. the power of labels